index.rb

```diff
@@ -0,0 +1,120 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    list = is_a?(Range) ? to_a : self
+    i = 0
+    while i < list.length
+      yield(list[i])
+      i += 1
+    end
+    list
+  end
+
+  def my_each_with_index()
+    return enum_for unless block_given?
+
+    list = is_a?(Range) ? to_a : self
+    index = 0
+    while index < list.length
+      yield(self[index], index)
+      index += 1
+    end
+  end
+
+  def my_select
+    return enum_for unless block_given?
+
+    filter = []
+    my_each { |element| filter.push(element) if yield(element) }
+    filter
+  end
+
+  def my_all?(*args)
+    lever = true
+    if !args[0].nil?
+      my_each { |element| lever = false unless args[0] == element }
+    elsif !block_given?
+      my_each { |element| lever = false unless element }
+    else
+      my_each { |element| lever = false unless yield(element) }
+    end
+    lever
+  end
+
+  def my_any?(*arg)
+    lever = false
+    if !arg[0].nil?
+      my_each { |element| lever = true if arg[0] == element }
+    elsif !block_given?
+      my_each { |element| lever = true if element }
+    else
+      my_each { |element| lever = true if yield(element) }
+    end
+    lever
+  end
+
+  def my_none?(arg = nil, &block)
+    !my_any?(arg, &block)
+  end
```

Comment:
- [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_none?

- [ ]  I see you get my_none? from my_any? so this method will be fixed whit the changes requested in my_any?